### PR TITLE
Reserve Names for Pandas Mapper

### LINKED ISSUE #4886: Add Reserved Names in Pandas Converter
#### Expected Behavior
Be access `low` column in pandas Dataframe with
```python
history = self.History(symbol, 1, Resolution.Minute).low
```
#### Actual Behavior
The statement above raises the following exception:
```AttributeError : 'DataFrame' object has no attribute 'LOW R735QTJ8XC9X'```

#### Potential Solution
Add reserved names in `PandasConverter` to not convert "low" into "LOW R735QTJ8XC9X". We will need to use "LOW" instead.

#### Reproducing the Problem
```python
class NadionDynamicChamber(QCAlgorithm):
    def Initialize(self):
        self.SetStartDate(2020, 1, 6)
        symbol = self.AddEquity('LOW').Symbol
        history = self.History(symbol, 1, Resolution.Minute)
        self.Quit(history.low)
```
#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by C-SELLERS**
#### Description

Force pandas mapper to not convert strings 'low' 'high' 'close' and 'open' to symbol before indexing dataframe. Now those keywords are reserved and won't attempt to be converted.

#### Related Issue

Closes #4886 

#### Motivation and Context

Pandas bug specific to stock ticker LOW, when requesting history for LOW and attempting to access history `.low` values pandas would convert the request for `.low` to symbol LOW which pandas dataframe does not have a value for thus causing the issue.

I reserved more than just low since it is possible that other tickers may fit some of those column names in the future. 

#### Requires Documentation Change

N/A

#### How Has This Been Tested?

- Added a test that red/greened from master to this pr.
- Confirmed original behavior DataFrame.loc["LOW"] can return a dataframe just for LOW ticker
- Confirmed DataFrame.low is now working

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Review by AlexCatarino — CHANGES_REQUESTED** [review]
Thank you, @C-SELLERS .
I think we should add a test that throws an exception for completeness. For example, what if the user tries to access the "low" (lower case) key? If should throw if we are referring to the Symbol, and should not if referring to the field.

File: Tests/Python/PandasConverterTests.cs [hunk 1]
```diff
@@ -223,6 +223,47 @@ public void HandlesNullableValues()
...
+                dynamic test = PythonEngine.ModuleFromString("testModule",
+    $@"
+def Test(dataFrame):
+    data = dataFrame.loc['LOW']
+    if data.empty:
+        raise Exception('LOW history data is empty')
+    if data.__len__() != 10:
+        raise Exception('Expected 10 data points')
+    lowData = data.low
+    if data.empty is 0:
```
Comment (Martin-Molinero) [inline Tests/Python/PandasConverterTests.cs:259]: `if data.empty is 0` should check `lowData` instead?

Comment (C-SELLERS) [reply Tests/Python/PandasConverterTests.cs:259]: My bad, thanks boss!!!!

**Review by AlexCatarino — APPROVED** [review]
Thank you for the extra tests! 👍

File: Tests/Python/PandasConverterTests.cs [hunk 2]
```diff
@@ -223,6 +223,97 @@ public void HandlesNullableValues()
...
+def Test2(dataFrame):
+    # Bad accessor, expected to throw
+    data = dataFrame.LOW
+def Test3(dataFrame):
+    # Bad key, expected to throw
+    data = dataFrame.loc['low']
+def Test4(dataFrame):
+    # Should not throw, access data column low for all tickers
+    data = dataFrame.low
+");
```
Comment (Martin-Molinero) [inline Tests/Python/PandasConverterTests.cs:302]: Sweet test! 👍

**Review by Martin-Molinero — APPROVED** [review]

**Review by Martin-Molinero — APPROVED** [review]

**Pull request merged**